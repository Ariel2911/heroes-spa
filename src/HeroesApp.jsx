export const HeroesApp = () => {
  return (
    <>
      <h1>App</h1>
      <p>akjfhdjh</p>
    </>
  );
};
