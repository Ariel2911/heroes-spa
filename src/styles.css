:root {
  --primari-color: #4661fa;
  --primari-color-alpha: #4661fa34;
  --primari-color-ligth: #c7f4fc;
  --primari-color-dark: #395e6e;
  --danger-color: #e61213;
  --danger-color-alpha: #e6121257;
  --grey-color: #808080;
}

body {
  margin: 0;
  background-color: var(--primari-color-dark);
  color: var(--primari-color-ligth);
}

a {
  text-decoration: none;
}

*,
::after,
::before {
  box-sizing: border-box;
}

.navbar {
  position: sticky;
  top: 0;

  display: flex;
  justify-content: start;
  align-items: center;
  gap: 8px;
  padding: 4px;

  background-color: var(--primari-color-ligth);
}

.navbar-collapse {
  align-items: center;
}

.navbar-nav {
  margin: 0;
  padding: 0;
  list-style: none;
}

.nav-item {
  margin: 8px;
}

.navbar-brand {
  font-size: 1.25rem;
  font-weight: 600;
}

.d-flex {
  display: flex;
  justify-content: end;
  flex-grow: 1;
}

.text-prymary {
  color: var(--primari-color);
  font-weight: 600;
}

.btn {
  cursor: pointer;
  background-color: transparent;
  border: none;
  font-size: inherit;
}

.btn-prymary {
  background-color: var(--primari-color);
  border-radius: 4px;
  border: 1px solid black;
  color: var(--primari-color-ligth);
  padding: 4px 8px;
}

.active {
  color: red;
}

.container {
  margin: 0 52px;
}

.row {
  display: flex;
  flex-wrap: wrap;
  gap: 16px
}

.card {
  width: 216px;
}

.card-img {
  width: 100%;
}

.img-thumbnail {
  width: 60vw;
}

.alert {
  width: 70vw;
  padding: 8px;
  margin: 8px 0;
  border: 1px solid var(--grey-color);
  border-radius: 4px;
}

.alert-primary {
  background-color: var(--primari-color-alpha);
}

.alert-danger {
  background-color: var(--danger-color-alpha);
}