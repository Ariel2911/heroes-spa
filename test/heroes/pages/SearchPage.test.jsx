import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter, Navigate } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Pruebas en <SearchPage />', () => {
  beforeEach(() => jest.clearAllMocks());

  test('debe mostrarse correctamente con valores por defecto', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test('debe mostrar a Batman y el input con el valor del queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    );

    // screen.debug();

    const input = screen.getByRole('textbox');

    expect(input.value).toBe('batman');

    const img = screen.getByRole('img');

    expect(img.src).toContain('/heroes/dc-batman.jpg');

    const alertDanger = screen.getByLabelText('alert-danger');

    expect(alertDanger.style.display).toBe('none');
  });

  test('debe mostrar un error si no se encuentra el heroe (batman123)', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman123']}>
        <SearchPage />
      </MemoryRouter>
    );

    // screen.debug();

    const alertDanger = screen.getByLabelText('alert-danger');

    expect(alertDanger.style.display).toBe('');
  });

  test('debe llamar el navigate a la pantalla nueva', () => {
    const inputValue = 'superman';
    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );

    // screen.debug();

    const input = screen.getByRole('textbox');

    fireEvent.change(input, {
      target: { name: 'searchText', value: inputValue },
    });

    const form = screen.getByLabelText('form');

    fireEvent.submit(form);

    expect(mockNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
  });
});
