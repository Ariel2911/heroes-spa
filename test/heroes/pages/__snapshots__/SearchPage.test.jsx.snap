// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h2>
    Search
  </h2>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h3>
        Searching
      </h3>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=" "
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h3>
        Result
      </h3>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
        style="display: none;"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
      >
        No hero with 
        <b>
           
        </b>
      </div>
    </div>
  </div>
</div>
`;
